File: agentguard/executors/mock_github.py
from agentguard.executors.base import Executor
from agentguard.models import ActionRequest


class MockGitHubExecutor(Executor):
    def preview(self, action_request: ActionRequest) -> dict:
        return {"executor": "mock_github", "summary": f"Would run {action_request.action_type} on {action_request.target_resource}"}

    def execute(self, action_request: ActionRequest, execution_token: str) -> dict:
        return {
            "executor": "mock_github",
            "executed": True,
            "message": f"Mock executed {action_request.action_type} on {action_request.target_resource}",
        }

File: ts-agentguard/src/executors/mock-github.ts
import { ActionRequest } from "../models";

export class MockGitHubExecutor {
  preview(actionRequest: ActionRequest): Record<string, unknown> {
    return {
      executor: "mock_github",
      summary: `Would run ${actionRequest.action_type} on ${actionRequest.target_resource}`,
    };
  }

  execute(actionRequest: ActionRequest, executionToken: string): Record<string, unknown> {
    void executionToken;
    return {
      executor: "mock_github",
      executed: true,
      message: `Mock executed ${actionRequest.action_type} on ${actionRequest.target_resource}`,
    };
  }
}
